clients/v1_dbrps/v1_dbrps.go

```diff
@@ -0,0 +1,246 @@
+		return clients.ErrMustSpecifyOrg
+	}
+
+	req := c.GetDBRPs(ctx)
+	if params.OrgID.Valid() {
+		req = req.OrgID(params.OrgID.String())
+	}
+	if params.OrgName != "" {
+		req = req.Org(params.OrgName)
+	}
+	if !params.OrgID.Valid() && params.OrgName == "" {
+		req = req.Org(c.ActiveConfig.Org)
+	}
+
+	if params.ID != "" {
+		req = req.Id(params.ID)
+	}
+	if params.BucketID != "" {
+		req = req.BucketID(params.BucketID)
+	}
+	if params.RP != "" {
+		req = req.Rp(params.RP)
+	}
+	if params.DB != "" {
+		req = req.Db(params.DB)
+	}
+
+	// Set this parameter if the --default flag was passed. This will list only
+	// default DBRPs. Otherwise, don't set the parameter at all to list DBRPs that
+	// are default and not default.
+	if params.Default_ {
+		req = req.Default_(params.Default_)
+	}
+
+	dbrps, err := req.Execute()
+	if err != nil {
+		return fmt.Errorf("failed to list dbrps: %w", err)
+	}
+
+	c.printDBRPs(dbrpPrintOpts{dbrps: dbrps.GetContent()})
+
+	return nil
+}
+
+type CreateParams struct {
+	clients.OrgParams
+	BucketID string
+	Default_ bool
+	DB       string
+	RP       string
+}
+
+func (c Client) Create(ctx context.Context, params *CreateParams) error {
+	if !params.OrgID.Valid() && params.OrgName == "" && c.ActiveConfig.Org == "" {
+		return clients.ErrMustSpecifyOrg
+	}
+
+	reqBody := api.DBRPCreate{
+		BucketID:        params.BucketID,
+		Database:        params.DB,
+		RetentionPolicy: params.RP,
+	}
+
+	// For compatibility with the cloud API for creating a DBRP, an org ID must be
+	// provided. The ID will be obtained based on the org name if an
+	// org name is provided but no ID is.
+	if params.OrgID.Valid() {
+		reqBody.OrgID = api.PtrString(params.OrgID.String())
+	} else {
+		orgName := params.OrgName
+		if orgName == "" {
+			orgName = c.ActiveConfig.Org
+		}
+		res, err := c.GetOrgs(ctx).Org(orgName).Execute()
+		if err != nil {
+			return fmt.Errorf("failed to look up ID for org %q: %w", orgName, err)
+		}
+		if len(res.GetOrgs()) == 0 {
+			return fmt.Errorf("no org found with name %q", orgName)
+		}
+		reqBody.OrgID = api.PtrString(res.GetOrgs()[0].GetId())
+	}
+
+	dbrp, err := c.PostDBRP(ctx).DBRPCreate(reqBody).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to create dbrp for bucket %q: %w", params.BucketID, err)
+	}
+
+	c.printDBRPs(dbrpPrintOpts{dbrp: &dbrp})
+
+	return nil
+}
+
+type UpdateParams struct {
+	clients.OrgParams
+	ID       string
+	Default_ bool
+	RP       string
+}
+
+func (c Client) Update(ctx context.Context, params *UpdateParams) error {
+	if !params.OrgID.Valid() && params.OrgName == "" && c.ActiveConfig.Org == "" {
+		return clients.ErrMustSpecifyOrg
+	}
+
+	reqBody := api.DBRPUpdate{}
+	if params.RP != "" {
+		reqBody.RetentionPolicy = &params.RP
+	}
+	if params.Default_ {
+		reqBody.Default = &params.Default_
+	}
+
+	req := c.PatchDBRPID(ctx, params.ID)
+	if params.OrgID.Valid() {
+		req = req.OrgID(params.OrgID.String())
+	}
+	if params.OrgName != "" {
```

Comment:
Does the cloud API support update-by-org-name? Or do we need to look up the org ID here as well?

Reply:
Updating with an org name does work for this. It's strange that creating with an org name doesn't work for cloud - I think it's probably a bug. I opened an issue in IDPE for that.